### DISCUSSION TITLE
How to ensure that we can deliver a stable WebView2 browser to customers?

### DISCUSSION THREAD

**Discussion by fvanheeswijk** [MicrosoftEdge/WebView2Feedback#4008] [Q&A]
We have recently considered several options, for example the Evergreen Runtime and Fixed Version runtime distribution options but we are yet to find one that fits our needs. Hence we are asking you for help. The current situation is that the WebView2 browser simply is not stable enough and we are trying to drop support of the old IE-based (WinForms) browser control as that one is also not stable enough and prevents us from developing newer features in our application.

Our application has several releases, ranging from a weekly/monthly in-development version to a version that is two years old. We are only able to update parts of the database of the older versions, and we cannot update the C++ and installer parts. More specifically, once a version gets released we cannot update to a newer (C#/.NET) SDK in that old version. If we were to use a Fixed Version runtime then we also are not able to update that one. If we would decide to use a Fixed Version runtime then the oldest versions would lack two years worth of bug- and security fixes, this is not acceptable to us.

Hence we decided to use the Evergreen Runtime, but fact is that Microsoft simply makes too many mistakes resulting in our application being broken on all of our versions simultaneously with no way to fix those issues by us, the only option is to wait for Microsoft to fix the problem. The result is that customers face an unstable application for at least one week on average, this is actually the current situation, and is yet again unacceptable.

So how do we ensure that we can maintain an application that is not subject to WebView2 issues?
Is it possible for Microsoft to change the way the Evergreen Runtime works such that the previous version is also always available and we can always fallback to that version from the (C#/.NET) SDK, via command line options for example?
Or do you have any other suggestions?

**Comment by nishitha-burman** [comment]
Hi @fvanheeswijk,

Thank you for the feedback. Can you please expand on the WebView2 issues you run into?

Thank you

**Comment by fvanheeswijk** [comment]
@nishitha-burman There are numerous issues we keep running into regularly, for example #3062, but that really is besides the point. We just want to have a stable browser component available at all times as explained in the original post.

**Comment by diffent** [comment]
Seeing that this is dated only 5 weeks ago, can anyone from MS vouch for WebView2 being ready for production use?  Is it stable enough, or are there breaking changes coming up in the near future?  The long discussion in #3062 noted above may indicate that WebView2 is not yet stable.  It is unclear to this new WebView2 trial user if these are production releases or beta though.
